SAWER Onboarding

sawer-odoo opened the pull request:
sawer chapter 2

sawer-odoo committed:
[ADD] Chapter 2 sawer

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/tutorials/pull/997.png)](https://mergebot.odoo.com/odoo/tutorials/pull/997)

sawer-odoo committed:
[ADD] Chapter 3 sawer

[ADD] Chapter 2 sawer - correction

sawer-odoo committed:
[ADD] Chapter 4 sawer

sawer-odoo committed:
[ADD] Chapter 5 sawer

[FIX] missing space correction

Mathilde411 commented on estate/security/ir.model.access.csv:
```diff
@@ -0,0 +1,2 @@
+id,name,model_id/id,group_id/id,perm_read,perm_write,perm_create,perm_unlink
+access_estate_property,access_estate_property,model_estate_property,base.group_user,1,1,1,1
```
End of file line is missing here

Mathilde411 commented on estate/views/estate_menus.xml:
```diff
@@ -0,0 +1,15 @@
+
+    </menuitem>
+
+</odoo>
```
End of file line is missing here

Mathilde411 commented on estate/views/estate_property_views.xml:
```diff
@@ -0,0 +1,15 @@
+        </field>
+    </record>
+
+</odoo>
```
End of file line is missing here

Mathilde411 commented on estate/models/estate_property.py:
```diff
@@ -0,0 +1,47 @@
+    description = fields.Text(string='Description')
+    postcode = fields.Char(string='Post code')
+    date_availability = fields.Date(
+        string='Availability date',
+        default=fields.Date.add(fields.Date.today(), months=3),
+        copy=False 
+    )
+    expected_price = fields.Float(string='Expected price', required=True)
+    selling_price = fields.Float(
+        string='Selling price',
+        readonly=True,
+        copy=False
+    )
+    bedrooms = fields.Integer(
+        string='Bedrooms',
+        default=2
+    )
+    living_area = fields.Integer(string='Living area')
+    facades = fields.Integer(string='Facades')
+    garage = fields.Boolean(string='Garage')
+    garden = fields.Boolean(string='Garden')
+    garden_area = fields.Integer(string='Garden area')
+    garden_orientation = fields.Selection( 
+        string='Garden orientation',
+        selection= [('north', 'North'), ('south', 'South'), ('east', 'East'), ('west', 'West')]
+    )
+    active = fields.Boolean(
+        string='Active',
+        default=True
+    )
+    state = fields.Selection(
+        string='State',
+        selection= [('new', 'New'), ('offer_received', 'Offer Received'), ('offer_accepted', 'Offer Accepted'), ('sold', 'Sold'), ('cancelled', 'Cancelled')],
+        default='new'
+    )
```
```suggestion
    name = fields.Char(required=True)
    description = fields.Text()
    postcode = fields.Char()
    date_availability = fields.Date(
        string='Availability date',
        default=fields.Date.add(fields.Date.today(), months=3),
        copy=False 
    )
    expected_price = fields.Float(required=True)
    selling_price = fields.Float(
        readonly=True,
        copy=False
    )
    bedrooms = fields.Integer(default=2)
    living_area = fields.Integer()
    facades = fields.Integer()
    garage = fields.Boolean()
    garden = fields.Boolean()
    garden_area = fields.Integer()
    garden_orientation = fields.Selection( 
        selection= [('north', 'North'), ('south', 'South'), ('east', 'East'), ('west', 'West')]
    )
    active = fields.Boolean(default=True)
    state = fields.Selection(
        selection= [('new', 'New'), ('offer_received', 'Offer Received'), ('offer_accepted', 'Offer Accepted'), ('sold', 'Sold'), ('cancelled', 'Cancelled')],
        default='new'
    )
```
Here the string are correctly generated from the variable names,no need for them :)

Mathilde411 commented on estate/models/estate_property.py:
```diff
@@ -0,0 +1,47 @@
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "Create a new table"
```
Description should be a human readable name for you model.
e.g. for model with `_name = 'sale.order'` we have `_description = "Sales Order"`

Mathilde411 reviewed (changes requested):
Small comments :)

sawer-odoo committed:
[ADD] Chapter 6 sawer

sawer-odoo committed:
[ADD] Chapter 7 sawer

sawer-odoo committed:
[ADD] Chapter 8 sawer

[FIX] estate.propety.offer_id existence verification

[FIX] trailing white space in estate_property_offer: 34

sawer-odoo committed:
[ADD] Chapter 9 sawer

[FIX] missing files in button definition

[FIX] using unknown field "states" from copy paste

sawer-odoo committed:
[ADD] Chapter 10 sawer

[FIX] blank spaces

[FIX] whitespace arround operator

[IMP] estate: Add notes.

sawer-odoo committed:
[ADD] Chapter 11 sawer

[FIX] definition order in manifest.

sawer-odoo committed:
[ADD] Chapter 12 sawer

[FIX] erased the print

[ADD] Unittest offer creation (chapter 12)

sawer-odoo committed:
[ADD] Chapter 13 sawer

[FIX] style fix for runbot

sawer-odoo committed:
[ADD] Chapter 14 sawer

sawer-odoo committed:
[IMP] Chapter 15 sawer - file renaming

[IMP] modify garden onchange condition

sawer-odoo committed:
[ADD] unittest property selling + garden fields reset

[FIX] runbot style issues

sawer-odoo committed:
[ADD]  Define module data

sawer-odoo committed:
[ADD] Restrict access to data

sawer-odoo committed:
[FIX] runbot fix : manifest declaration order + print

sawer-odoo committed:
[FIX] manifest declaration order

sawer-odoo committed:
[FIX] style issue
